test/exercise/fp2/solution.rb

```diff
@@ -6,18 +6,33 @@ class MyArray < Array
 
       # Написать свою функцию my_each
       def my_each
+        i = 0
+        while i < count
+          yield(self[i])
+          i += 1
+        end
+        self
       end
 
       # Написать свою функцию my_map
       def my_map
+        result = MyArray.new([])
+        my_each { |el| result << yield(el) }
+        result
       end
 
       # Написать свою функцию my_compact
       def my_compact
+        result = MyArray.new([])
+        my_each { |el| result << el unless el.nil? }
```

Comment:
it's better not to use 'unless' once we have 'present'.
It's extended in rails, but anyway

Reply:
Fixed. 
Yep, remember reading about it in the theory.
Should I just ignore the lint offence then? It says "Favor unless over if for negative conditions."

Reply:
Ignore it for now, it will never happen in real Rails project
or you can write
`my_each { |el| result << el if el }`

Reply:
I tried changing it to `my_each { |el| result << el if el }` like you suggested, but it removes all `false` values from array, when this check is executed:
```
func_yet_another = -> (element) { element.even? }
assert @array.map(&func_yet_another).compact == @my_array.my_map(&func_yet_another).my_compact
```
So I'll leave it the way it is for now.
Thanks for the review!

Reply:
Yes, it makes sense